Kata RPG/Kata RPG/Warrior.swift

```diff
@@ -9,37 +9,29 @@ import Foundation
 
 class Warrior: Character {
    
-    override func attack(target: Character){
-            if(self.lifeStatus == true){
+     func attack(target: Character){
+        if(self.lifeStatus == true && (self.faction !== target.faction && self.faction != nil && target.faction != nil )){
```

Comment:
Same comment as for priest. You shouldn't have had to do that imho, needs factorization